csrc/python_frontend/fusion_definition.cpp

```diff
@@ -368,20 +373,29 @@ Tensor FusionDefinition::addTensor(TensorView* tv) {
 
 Tensor FusionDefinition::defineTensor(size_t dims) {
   FUSER_PERF_SCOPE("FusionDefinition::defineTensor");
+  NVF_CHECK(
+      trie_node_ != nullptr,
+      "defineTensor must be called inside a 'with' block");
   Tensor out(recording_state_.size(), dims, this);
   recording_state_.emplace_back(out(), serde::StateType::Tensor);
   return out;
 }
 
 Vector FusionDefinition::defineVector(size_t size) {
   FUSER_PERF_SCOPE("FusionDefinition::defineVector");
+  NVF_CHECK(
+      trie_node_ != nullptr,
+      "defineVector must be called inside a 'with' block");
   Vector out(recording_state_.size(), size, this);
   recording_state_.emplace_back(out(), serde::StateType::Vector);
   return out;
 }
 
 void FusionDefinition::defineRecord(RecordFunctor* record) {
   FUSER_PERF_SCOPE("FusionDefinition::defineRecord");
+  NVF_CHECK(
+      trie_node_ != nullptr,
+      "defineRecord must be called inside a 'with' block");
```

Comment:
```suggestion
      "defineRecord() must be called from an initialized definition via a python context manager or a child class' definition() method.");
```

Reply:
I don't think this method is actually exposed via python but it doesn't hurt to add the check.